### PR TITLE
feat(getContent): return item group ids

### PR DESCRIPTION
**Pull request by tomwayson**
affects: @esri/hub-common,  @esri/hub-content

continuation of the effort to address: #322

When in portal make an extra fetch to get the item's groups.

### CODE REVIEW

File: packages/content/src/portal.ts [hunk 1]
```diff
@@ -16,10 +16,61 @@ import {
...
+    const { admin, member, other } = response;
+    return [...admin, ...member, ...other].map(group => group.id);
+  });
+}
+
+// simultaneously execute multiple async requests to populate content properties
+// returns a hash of all the resolved property values that also has
+// an array of property names for which the request failed
+// NOTE: assumes each request takes only the item id and request options
+function fetchProperties(
```
Comment (tomwayson) [inline packages/content/src/portal.ts:38]: this function makes it easy to add more fetches for properties like `server` that we'll need to populate for datasets.

File: packages/content/src/portal.ts [hunk 2]
```diff
@@ -16,10 +16,61 @@ import {
...
+        const propertyName = propertyNames[i];
+        if (value === propertyName) {
+          // capture that the request failed
+          properties.errors.push(propertyName);
+        } else {
+          properties[propertyName] = value;
+        }
+        return properties;
+      },
+      { errors: [] } as { [key: string]: unknown }
```
Comment (tomwayson) [inline packages/content/src/portal.ts:69]: leaving this undocumented for now until I verify the format of the errors returned by the Hub REST API

Comment (drewctate) [reply packages/content/src/portal.ts:69]: Leave a `TODO` comment to this effect perhaps?

File: packages/content/src/portal.ts (hunk 1 above)
Comment (drewctate) [reply packages/content/src/portal.ts:38]: Nice API design.

**Review by drewctate — APPROVED** [review]
Nice work.

File: packages/content/src/portal.ts [hunk 3]
```diff
@@ -45,29 +45,34 @@ function fetchProperties(
...
-  );
+  const requests = propertyNames.map(propertyName => {
+    // initiate the request and return the promise
+    const request = propertyRequests[propertyName];
+    return request(itemId, requestOptions).catch(e => {
+      // there was an error w/ the request, capture it
+      const message = (e && e.message) || e;
+      errors.push({
+        // NOTE: for now we just return the message and type "Other"
+        // but we could later introspect for HTTP or AGO errors
```
Comment (drewctate) [inline packages/content/src/portal.ts:58]: 🤔 trying to think of a use-case for this.

Comment (drewctate) [reply packages/content/src/portal.ts:58]: Never mind. I see that that's just what the indexer does.

File: packages/common/src/types.ts [hunk 4]
```diff
@@ -288,6 +297,11 @@ export interface IHubContent extends IHubResource, IItem {
   portalDataUrl?: string;
   /** The ids of any groups that the item belongs to */
   groupIds?: string[];
+  /**
+   * Any errors encountered when indexing or composing the content
+   * see https://github.com/ArcGIS/hub-indexer/blob/dc6146c3b7b4007b530f8e6357a9c36db59a6996/docs/errors.md#response-formatting-for-errors
```
Comment (drewctate) [inline packages/common/src/types.ts:302]: Maybe change to be the `master` link? `https://github.com/ArcGIS/hub-indexer/blob/master/docs/errors.md#response-formatting-for-errors`

**Review by drewctate — APPROVED** [review]
LGTM

**Pull request merged**